search/client.go

```diff
@@ -110,3 +109,25 @@ func (c *client) ListFacetValues(field string, maximumNumberOfValues int) (*Face
 	err = json.NewDecoder(resp.Body).Decode(facetValues)
 	return facetValues, nil
 }
+
+func createRequest(client *client, reqType string, queryStringParams map[string]string, customHeaders map[string]string, endpoint string, body io.Reader) (*http.Request, error) {
+	req, err := http.NewRequest(reqType, endpoint, body)
+	if err != nil {
+		return nil, err
+	}
+
+	customHeaders["Authorization"] = "Bearer " + client.token
+	customHeaders["User-Agent"] = client.useragent
+	for key, value := range customHeaders {
```

Comment:
This is not more efficient, you are creating a new map, when you already have one.